types/extra-types.d.ts

```diff
@@ -0,0 +1,163 @@
+  observer?: string;
+}
+
+type PolymerElementProperties = {
+  [key: string]: PolymerElementPropertiesMeta
+};
+
+interface PolymerInit {
+  is: string;
+  extends?: string;
+  properties?: PolymerElementProperties;
+  observers?: string[];
+  template?: HTMLTemplateElement|string;
+  hostAttributes?: {[key: string]: any};
+  listeners?: {[key: string]: string};
+}
+
+/**
+ * Types from "externs/polymer-internal-shared-types.js"
+ */
+
+interface StampedTemplate extends DocumentFragment {
+  __noInsertionPoint: boolean;
+  nodeList: Node[];
+  $: {[key: string]: Node};
+  templateInfo?: TemplateInfo;
+}
+
+interface NodeInfo {
+  id: string;
+  events: {name: string, value: string}[];
+  hasInsertionPoint: boolean;
+  templateInfo: TemplateInfo;
+  parentInfo: NodeInfo;
+  parentIndex: number;
+  infoIndex: number;
+  bindings: Binding[];
+}
+
+interface TemplateInfo {
+  nodeInfoList: NodeInfo[];
+  nodeList: Node[];
+  stripWhitespace: boolean;
+  hasInsertionPoint?: boolean;
+  hostProps: Object;
+  propertyEffects: Object;
+  nextTemplateInfo?: TemplateInfo;
+  previousTemplateInfo?: TemplateInfo;
+  childNodes: Node[];
+  wasPreBound: boolean;
+}
+
+interface LiteralBindingPart {
+  literal: string;
+  compoundIndex?: number;
+}
+
+interface MethodArg {
+  literal: boolean;
+  name: string;
+  value: string|number;
+  rootProperty?: string;
+  structured?: boolean;
+  wildcard?: boolean;
+}
+
+interface MethodSignature {
+  methodName: string;
+  static: boolean;
+  args: MethodArg[];
+  dynamicFn?: boolean;
+}
+
+interface ExpressionBindingPart {
+  mode: string;
+  negate: boolean;
+  source: string;
+  dependencies: Array<MethodArg|string>;
+  customEvent: boolean;
+  signature: Object|null;
+  event: string;
+}
+
+type BindingPart = LiteralBindingPart|ExpressionBindingPart;
+
+interface Binding {
+  kind: string;
+  target: string;
+  parts: BindingPart[];
+  literal?: string;
+  isCompound: boolean;
+  listenerEvent?: string;
+  listenerNegate?: boolean;
+}
+
+/**
+ * Types from "lib/utils/async.html"
+ */
+
+interface AsyncInterface {
+  run: (fn: Function, delay?: number) => number;
+  cancel: (handle: number) => any;
+}
+
+/**
+ * Types from "lib/utils/debounce.html"
+ */
+
+type AsyncModule = AsyncInterface;
+
+/**
+ * Types from "lib/utils/gestures.html"
+ */
+
+interface GestureRecognizer {
+  reset: () => any;
+  mousedown?: (e: MouseEvent) => any;
+  mousemove?: (e: MouseEvent) => any;
+  mouseup?: (e: MouseEvent) => any;
+  touchstart?: (e: TouchEvent) => any;
+  touchmove?: (e: TouchEvent) => any;
+  touchend?: (e: TouchEvent) => any;
+  click?: (e: MouseEvent) => any;
+}
+
+/**
+ * Not defined in the TypeScript DOM library.
+ * See https://developer.mozilla.org/en-US/docs/Web/API/IdleDeadline
+ */
+interface IdleDeadline {
+  didTimeout: boolean;
+  timeRemaining(): number;
+}
+
+/**
+ * Closure type equivalence for tagged template literal function argument.
+ */
+type ITemplateArray = TemplateStringsArray;
```

Comment:
Ah, I didn't see this when I looked at `Polymer.html`. Can't you just add this to the Closure -> TS type converter functions?

Reply:
Fixed via https://github.com/Polymer/gen-typescript-declarations/pull/47.